Source/WebCore/platform/network/cf/DNSResolveQueueCFNet.cpp

```diff
@@ -165,10 +165,11 @@ void DNSResolveQueueCFNet::performDNSLookup(const String& hostname, Ref<Completi
 {
     ASSERT(isMainThread());
     DNSServiceRef service { nullptr };
-    DNSServiceErrorType result = DNSServiceGetAddrInfo(&service, kDNSServiceFlagsReturnIntermediates, kDNSServiceInterfaceIndexAny, kDNSServiceProtocol_IPv4 | kDNSServiceProtocol_IPv6, hostname.utf8().data(), dnsLookupCallback, &wrapper.leakRef());
+    auto& leakedWrapper = wrapper.leakRef();
```

Comment:
This is an unusual way to move an rvalue reference, but I guess it works looking at `Ref::leakRef()`.  Would be nice if we could use a plain `WTF::Move()` or `std::exchange()` instead to make it more obvious what's happening.

Won't `leakedWrapper` leak when `result == kDNSServiceErr_NoError`, though?  If so, please fix before landing.

Reply:
Or does `DNSServiceGetAddrInfo()` consume `leakedWrapper` in the non-error case?

Reply:
dnsLookupCallback re-adopts the wrapper in that case.  This code doesn't change that case.

Reply:
Yes, `dnsLookupCallback()` consumes `leakedWrapper` when it's called, apparently in the no-error case.